cesium/writer_stream.go

```diff
@@ -413,16 +412,14 @@ func (w *idxWriter) validateWrite(fr Frame) error {
 			continue
 		}
 
+		if fr.Series[i].DataType != uWriter.Channel.DataType {
+			return errors.Wrapf(
+				validate.Error,
+				"invalid data type for channel %d, expected %s, got %s",
+				k, uWriter.Channel.DataType, fr.Series[i].DataType)
+		}
 		if lengthOfFrame == -1 {
-			s := fr.Series[i]
-			if s.DataType.Density() == 0 {
```

Comment:
Can you explain this change?

Reply:
We want to check every frame for whether it has the right data type, not just the first.

Reply:
We want to check whether every series has the right data type, not just the first

Reply:
Shouldn't we be checking this in the unary layer? I added this so that getting the length of the series doesn't panic on a variable length series

Reply:
Yes, you're right, we're already checking at the unary level whether the data type is correct. But I still don't understand why we only chekc for the first channel if the data type is unknown. Could you explain what you mean by "variable length series". Isn't getting the length of the frame supposed to panic when the frame is not even?

Reply:
Also, just to confirm – at the unary level we check as if int64 and timestamp data type are interchangeable – should that be the case? Should we allow passing a int64 series into a timestamp channel?